From: Hannes Reinecke <hare@suse.de>
Date: Fri, 27 Nov 2015 10:52:08 +0100
Subject: [PATCH] block: Always check queue limits for cloned requests
References: bsc#902606
Patch-Mainline: v4.4
Git-commit: bf4e6b4e757488dee1b6a581f49c7ac34cd217f8

When a cloned request is retried on other queues it always needs
to be checked against the queue limits of that queue.
Otherwise the calculations for nr_phys_segments might be wrong,
leading to a crash in scsi_init_sgtable().

Acked-by: Mike Snitzer <snitzer@redhat.com>
Cc: Ewan Milne <emilne@redhat.com>
Cc: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-core.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/block/blk-core.c b/block/blk-core.c
index 801ced7..aa09f48 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -1928,9 +1928,6 @@ EXPORT_SYMBOL(submit_bio);
  */
 int blk_rq_check_limits(struct request_queue *q, struct request *rq)
 {
-	if (!rq_mergeable(rq))
-		return 0;
-
 	if (blk_rq_sectors(rq) > blk_queue_get_max_sectors(q, rq->cmd_flags)) {
 		printk(KERN_ERR "%s: over max size limit.\n", __func__);
 		return -EIO;
-- 
1.8.5.6

